_site/assets/js/global/mark-completed.js

```diff
@@ -1,15 +1,151 @@
 (function () {
   'use-strict';
-
   // retrieve the local storage "performanceKit" state object
   // localStorage.getItem('performanceKit') || {}
 
   // Get all the completeled buttons on the page
   // const completedButtons = [...document.querySelectorAll('[data-attribute-name]')];
   // const completedSections = [...document.querySelectorAll('[data-attribute-name]')];
-  // Each section will need a unique id in the html so go to /pathways/workshop/index.njk and you will see them there
+  // Each section will need a unique id in the html so go to /pathways/workshop/index.njk and you will see them there (18/1/22 - this is now walkthrough/index.njk)
   // Find which lessons are completed from localStorage store.completedLessons - should be an array of the lesson id's
   // loop through the lesson sections and assign a className if they are completed
   // loop through the buttons to assign a class and change the text for when it is completed
   // loop through the buttons and add an event listener for when they are clicked add the lesson to local storage as completed and change the button state to say "Lesson completed"
+
+  const init = () => {
+    addCompleteFlagToSections();
+    updateMarkCompleteButtonUI();
+    assignOnClickToButtons();
+  };
+
+  const assignOnClickToButtons = () => {
+    const buttons = getButtons();
+
+    for (let i = 0; i < buttons.length; i++) {
+      const button = buttons[i];
+      button.addEventListener('click', handleOnClick);
+    }
+  };
+
+  const updateMarkCompleteButtonUI = () => {
+    const buttons = getButtons();
+    const completedLessons = getStoreFromLocalStorage().completedLessons;
+
+    if (completedLessons && completedLessons.length) {
+      for (let i = 0; i < completedLessons.length; i++) {
+        const completedLesson = completedLessons[i];
+        for (let j = 0; j < buttons.length; j++) {
+          const button = buttons[i];
+          if (button.dataset.postid === completedLesson) {
+            button.classList.add('completed');
+            updateButton(button, 'add');
+          } else {
+            button.classList.remove('completed');
+            updateButton(button, 'remove');
+          }
+        }
+      }
+    }
+  };
+
+  const addCompleteFlagToSections = () => {
+    const sections = getSections();
+    const completedLessons = getStoreFromLocalStorage().completedLessons;
```

Comment:
Can do a cheeky `??` here where you do 

const completedLessons = getStoreFromLocalStorage().completedLessons ?? [];

also with the looping, it's a bit mental. You do this instead:

```
sections.forEach(section => {
  const isCompleted = completedLessons.find(lesson => lesson === section.id);

  if (isCompleted) {
    section.classList.add('completed');
    section.querySelector('[data-complete-tag]').classList.remove('invisible');
  } else {
     section.classList.remove('completed');
     section.querySelector('[data-complete-tag]').classList.add('invisible');
  }
});
```